test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs

```diff
@@ -20,83 +17,108 @@ public static class TestHelper
+            var provider = CallContextServiceLocator.Locator.ServiceProvider;
+            var originalEnvironment = provider.GetRequiredService<IApplicationEnvironment>();
 
             // When an application executes in a regular context, the application base path points to the root
             // directory where the application is located, for example MvcSample.Web. However, when executing
-            // an aplication as part of a test, the ApplicationBasePath of the IApplicationEnvironment points
+            // an application as part of a test, the ApplicationBasePath of the IApplicationEnvironment points
             // to the root folder of the test project.
             // To compensate for this, we need to calculate the original path and override the application
             // environment value so that components like the view engine work properly in the context of the
             // test.
-            var appBasePath = CalculateApplicationBasePath(appEnvironment, applicationWebSiteName, applicationPath);
-
-            var services = new ServiceCollection();
-            services.AddInstance(
-                typeof(IApplicationEnvironment),
-                new TestApplicationEnvironment(appEnvironment, appBasePath, applicationWebSiteName));
-
-            // Injecting a custom assembly provider via configuration setting. It's not good enough to just
-            // add the service directly since it's registered by MVC.
-            var providerType = CreateAssemblyProviderType(applicationWebSiteName);
-
-            var configuration = new TestConfigurationProvider();
-            configuration.Configuration.Set(
-                typeof(IAssemblyProvider).FullName,
-                providerType.AssemblyQualifiedName);
-
-            services.AddInstance(
-                typeof(ITestConfigurationProvider),
-                configuration);
-
-            services.AddInstance(
-                typeof(ILoggerFactory),
-                new LoggerFactory());
-
-            if (newServices != null)
-            {
-                services.Add(newServices);
-            }
-
-            return BuildFallbackServiceProvider(services, originalProvider);
+            var applicationBasePath = CalculateApplicationBasePath(
+                originalEnvironment,
+                applicationWebSiteName,
+                applicationPath);
+            var environment = new TestApplicationEnvironment(
+                originalEnvironment,
+                applicationBasePath,
+                applicationWebSiteName);
+            services.AddInstance<IApplicationEnvironment>(environment);
+
+            // Injecting a custom assembly provider. Overrides AddMvc() because that uses TryAdd().
+            var assemblyProvider = CreateAssemblyProvider(applicationWebSiteName);
+            services.AddInstance<IAssemblyProvider>(assemblyProvider);
+
+            services.AddInstance<ILoggerFactory>(new LoggerFactory());
```

Comment:
could rely on the one `AddHosting()` provides but that has a comment about being removed.  so, I'm not sure but don't think it matters much at this time.